cdap-tms/src/main/java/io/cdap/cdap/messaging/service/CoreMessagingService.java

```diff
@@ -87,22 +93,28 @@ public class CoreMessagingService extends AbstractIdleService implements Messagi
   private final long txMaxLifeTimeInMillis;
 
   @Inject
-  protected CoreMessagingService(CConfiguration cConf, TableFactory tableFactory,
+  protected CoreMessagingService(
+      CConfiguration cConf,
+      TableFactory tableFactory,
       MetricsCollectionService metricsCollectionService) {
     this(cConf, tableFactory, TimeProvider.SYSTEM_TIME, metricsCollectionService);
   }
 
   @VisibleForTesting
-  CoreMessagingService(CConfiguration cConf, TableFactory tableFactory,
-      TimeProvider timeProvider, MetricsCollectionService metricsCollectionService) {
+  CoreMessagingService(
+      CConfiguration cConf,
+      TableFactory tableFactory,
+      TimeProvider timeProvider,
+      MetricsCollectionService metricsCollectionService) {
     this.cConf = cConf;
     this.tableFactory = tableFactory;
     this.topicCache = createTopicCache();
     this.messageTableWriterCache = createTableWriterCache(true, cConf);
     this.payloadTableWriterCache = createTableWriterCache(false, cConf);
     this.timeProvider = timeProvider;
 
-    // Due to circular dependency in our class hierarchy (which is bad), we cannot use metricsCollectionService
+    // Due to circular dependency in our class hierarchy (which is bad), we cannot use
+    // metricsCollectionService
```

Comment:
I am wondering if this comment is true anymore?

Reply:
MessagingMetricsCollectionService does depend on MessagingService which depends on MetricsCollectionService . So I'd say it's still there.

Reply:
MessageingService is in SPI, so it should not depend in MetricsCollectionService. Implemnetation depends, but not spi

Reply:
Correct, that's what I meant.

Reply:
So, there is no circular dependency then, is it?